notesapi/v1/views.py

```diff
@@ -1,49 +1,86 @@
+from django.core.exceptions import ValidationError
 
 from rest_framework import status
 from rest_framework.response import Response
 from rest_framework.views import APIView
 
-from annotator.annotation import Annotation
+from notesapi.v1.models import Note
+
+if not settings.ES_DISABLED:
+    from notesserver.highlight import SearchQuerySet
 
-CREATE_FILTER_FIELDS = ('updated', 'created', 'consumer', 'id')
-UPDATE_FILTER_FIELDS = ('updated', 'created', 'user', 'consumer')
+log = logging.getLogger(__name__)
 
 
 class AnnotationSearchView(APIView):
     """
     Search annotations.
     """
-
     def get(self, *args, **kwargs):  # pylint: disable=unused-argument
         """
-        Search annotations.
+        Search annotations in most appropriate storage
+        """
+        # search in DB when ES is not available or there is no need to bother it
+        if settings.ES_DISABLED or 'text' not in self.request.QUERY_PARAMS.dict():
+            results = self.get_from_db(*args, **kwargs)
+        else:
+            results = self.get_from_es(*args, **kwargs)
+        return Response({'total': len(results), 'rows': results})
 
-        This method supports the limit and offset query parameters for paging
-        through results.
+    def get_from_db(self, *args, **kwargs):  # pylint: disable=unused-argument
+        """
+        Search annotations in database
         """
         params = self.request.QUERY_PARAMS.dict()
+        query = Note.objects.filter(
+            **{f: v for (f, v) in params.items() if f in ('course_id', 'usage_id')}
+        ).order_by('-updated')
 
-        if 'offset' in params:
-            kwargs['offset'] = _convert_to_int(params.pop('offset'))
-
-        if 'limit' in params and _convert_to_int(params['limit']) is not None:
-            kwargs['limit'] = _convert_to_int(params.pop('limit'))
-        elif 'limit' in params and _convert_to_int(params['limit']) is None:  # bad value
-            params.pop('limit')
-            kwargs['limit'] = settings.RESULTS_DEFAULT_SIZE
-        else:
-            # default
-            kwargs['limit'] = settings.RESULTS_DEFAULT_SIZE
+        if 'user' in params:
+            query = query.filter(user_id=params['user'])
 
-        # All remaining parameters are considered searched fields.
-        kwargs['query'] = params
+        if 'text' in params:
+            query = query.filter(text__icontains=params['text'])
 
-        results = Annotation.search(**kwargs)
-        total = Annotation.count(**kwargs)
+        return [note.as_dict() for note in query]
 
-        return Response({'total': total, 'rows': results})
+    def get_from_es(self, *args, **kwargs):  # pylint: disable=unused-argument
+        """
+        Search annotations in ElasticSearch
+        """
+        params = self.request.QUERY_PARAMS.dict()
+        query = SearchQuerySet().models(Note).filter(
+            **{f: v for (f, v) in params.items() if f in ('user', 'course_id', 'usage_id', 'text')}
```

Comment:
ah thanks, now i see.  so every request must have a user id param, and it must always match the token's, correct?  (other case to worry about is a user fetching notes for all users, by omitting the parameter from the request.)